sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxyUtil.java

```diff
@@ -0,0 +1,102 @@
+package com.azure.core.http.rest;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.implementation.http.UnexpectedExceptionInformation;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RestProxyUtil {
```

Comment:
Let's make this package-private, or if this needs to be public to be accessible across package boundaries let's move this into implementation.